lib/AST/Decl.cpp

```diff
@@ -6473,6 +6473,13 @@ static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
   auto baseInterfaceTy = base->getInterfaceType();
   auto derivedInterfaceTy = decl->getInterfaceType();
 
+  auto sig = ctx.getOverrideGenericSignature(base, (ValueDecl *)decl);
+
+  if (sig &&
+      !sig->requirementsNotSatisfiedBy(decl->getGenericSignature()).empty()) {
+    return true;
```

Comment:
Cool! Please add some SILGen tests too. Take a look at test/SILGen/vtable_thunks.swift and test/SILGen/vtables_thunk_reabstraction.swift to see how we test this sort of thing.

Reply:
@slavapestov I have added a test case!